doc/user/content/sql/recursive-ctes.md

```diff
@@ -0,0 +1,297 @@
+ERROR:  Evaluation error: Recursive query exceeded the recursion limit 100. (Use RETURN AT RECURSION LIMIT to not error, but return the current state as the final result when reaching the limit.)
+```
+
+The recursive CTE `reachable` has not converged to a fixpoint within the first 100 iterations!
+To see why, you can run variants of the same query where the
+
+```sql
+ERROR AT RECURSION LIMIT 100
+```
+
+clause is replaced by
+
+```sql
+RETURN AT RECURSION LIMIT $n -- where $n = 1, 2, 3, ...
+```
+
+and observe how the result changes after `$n` iterations.
+
+After 1 iteration:
+```text
+ src_id | dst_id
+--------+--------
+ A      | B
+ ...
+```
+
+After 2 iterations:
+```text
+ src_id | dst_id
+--------+--------
+ A      | B
+ A      | B
+ ...
+```
+
+After 3 iterations:
+```text
+ src_id | dst_id
+--------+--------
+ A      | B
+ A      | B
+ A      | B
+ ...
+```
+
+Changing the `UNION` to `UNION ALL` in the `reachable` definition caused a full copy of `transfer` to be added to the current value of `reachable` in each iteration!
+Consequently, `reachable` never stops growing and the recursive CTE computation never reaches a fixpoint.
+
+### Queries with "update locality"
+
+The examples presented so far have the following "update locality" property:
+
+{{< note >}}
+A change in a source collection will usually cause a _bounded amount_ of changes to the contents of the recursive CTE bindings derived after each iteration.
+{{</ note >}}
+
+For example:
+
+- Most of the time, inserting or removing `transfers` will not change the contents of the `reachable` collection. This is true because:
+  - An alternative path from `x` to `y` already existed before the insertion, or
+  - An alternative path from `x` to `y` will exist after the removal.
+- Inserting or removing `transfers` will not change most of the contents of the `area_balances` collection. This is true because:
+  - Areas are organized in a hierarchy with a maximum height `h`.
+  - A single transfer contributes directly only to the `area_balances` of the areas where the `src_id` and `tgt_id` belong.
+  - A single transfer contributes indirectly only to the `area_balances` of the areas ancestor areas of the above two areas.
+  - Consequently, each `transfer` change will affect at most `2` areas in each iteration and at most `2*h` areas in total.
+
+However, note that not all iterative algorithms have this property. For example:
+
+- In a naive PageRank implementation, the introduction of a link between two pages `x` and `y` will change the PageRank of these two pages and every page `z` transitively connected to either `x` or `y`.
+  Since most graphs exhibit [small-world properties](https://en.wikipedia.org/wiki/Small-world_network), this might represent most of the existing pages.
+- In a [naive k-means clustering algorithm](https://en.wikipedia.org/wiki/K-means_clustering), inserting or removing a new data point will effect the positions of the `k` mean points after each iteration.
```

Comment:
effect -> affect

https://www.merriam-webster.com/words-at-play/affect-vs-effect-usage-difference

Reply:
I'm now confused because @tr0njavolta requested the change in the opposite direction :)

Reply:
My bad! Read a little too quickly 😨